### DISCUSSION TITLE
varsIgnorePattern to ignore unused class names

### DISCUSSION THREAD

**Discussion by erosman** [eslint/eslint#17743] [Q&A]
## 1
How can an ignore be set for the unused class names (there are a few) used in Side Effect modules?

```js
class Something {

  static {
    this.set();
  }

  static set() {
    // do something
  }
}
```

.eslintrc.json
```json
"no-unused-vars": ["warn", {"varsIgnorePattern": ""}],
```

## 2
What is the regexp format used in "varsIgnorePattern"?

**Comment by Rec0iL99** [comment]
Hi @erosman, the choice is yours! You can pass in any regexp pattern that works for the `varsIgnorePattern` option.

for example,

```js
/* eslint "no-unused-vars": ["warn", {"varsIgnorePattern": "^_My"}] */

class _MySomething {

  static {
    this.set();
  }

  static set() {
    // do something
  }
}
```

Here all class names that start with `_My` will not be checked by the `no-unused-vars` rule.

> What is the regexp format used in "varsIgnorePattern"?

Any regexp pattern that is accepted by `new RegExp()`.

https://github.com/eslint/eslint/blob/05d6e99153ed6d94eb30f46c57609371918a41f3/lib/rules/no-unused-vars.js#L110

**Reply by erosman** [reply]
I understand. That would entail prefixing all of those classes in side effect modules. I share modules in various projects and prefer to avoid having to rename them.

I was hoping that there would be an option to ignore unused classes in "no-unused-vars".

I posted the following request, in case it could be useful:
 [Rule Change: no-unused-vars: add ignore by type option](https://github.com/eslint/eslint/issues/17757)

**Reply by Rec0iL99** [reply]
Ah ok, I misunderstood the question at first.
